enterprise/decorators.py

```diff
@@ -129,10 +129,13 @@ def wrapper(request, *args, **kwargs):
         enterprise_customer = get_enterprise_customer_or_404(enterprise_uuid)
         query_params = request.GET
         # Check if tpa_hint was passed as query param
-        tpa_hint = query_params.get('tpa_hint')
-        if not tpa_hint and not enterprise_customer.has_multiple_idps:
-            tpa_hint = enterprise_customer.identity_provider\
-                if enterprise_customer.identity_provider else None
+        tpa_hint_param = query_params.get('tpa_hint')
```

Comment:
Can we please add a comment here. So that it can be understandable why we are doing it.

Reply:
is not it self-explanatory?

Reply:
I think a comment that provides context would be helpful to future collaborators.

Reply:
moved this logic to a class method with comments.